dev/PowerNotifications/PowerNotifications.h

```diff
@@ -206,42 +237,80 @@ namespace winrt::Microsoft::Windows::System::Power
 
             event_token AddCallback(PowerFunctionDetails fn, const PowerEventHandler& handler)
             {
-                auto& eventObj{ fn.event() };
-                std::scoped_lock<std::mutex> lock(m_mutex);
-                if (!RegisteredForEvents(eventObj))
+                try
                 {
-                    fn.registerListener();
+                    PowerNotificationsTelemetry::AddCallbackTrace(fn.name.c_str());
+                    auto& eventObj{ fn.event() };
```

Comment:
What's motivating all these try/catch handlers? My estimation is that this should be moved out of here so there's a noexcept method like:

```c++
void PowerNotificationsTelemetry::AddCallbackTrace(PowerFunctionDetails const& fn) noexcept try
{
 
}
catch(...)
{
    FailureTrace(/* whatever */)
}
```

So the code here remains just `PowerNotificationsTelemetry::AddCallbackTrace(fn)` ... put the error handling where it belongs (in the place that generates errors) and not here.

Reply:
Kept only one instance.